modules/nf-core/evigene/tr2aacds/main.nf

```diff
@@ -0,0 +1,104 @@
+process EVIGENE_TR2AACDS {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/evigene:23.7.15--hdfd78af_1':
+        'biocontainers/evigene:23.7.15--hdfd78af_1' }"
+
+    input:
+    tuple val(meta), path(fasta)
+
+    output:
+    tuple val(meta), path("dropset")    , emit: dropset
+    tuple val(meta), path("okayset")    , emit: okayset
+    path "versions.yml"                 , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args        = task.ext.args     ?: ''
+    def prefix      = task.ext.prefix   ?: "${meta.id}"
+
+    def max_memory  = 7*1024
+    if (!task.memory) {
+        log.info '[evigene] Available memory not known - defaulting to 7GB. Specify process memory requirements to change this.'
+    } else {
+        max_memory  = (task.memory.mega*0.8).intValue()
+    }
+
+    def simple_name = fasta.simpleName
+    """
+    # Nextflow changes the container --entrypoint to /bin/bash (container default entrypoint: /usr/local/env-execute)
+    # Check for container variable initialisation script and source it.
+    if [ -f "/usr/local/env-activate.sh" ]; then
+        set +u  # Otherwise, errors out because of various unbound variables
+        . "/usr/local/env-activate.sh"
+        set -u
+    fi
+
+    \$EVIGENEHOME/scripts/prot/tr2aacds.pl \\
+        $args \\
+        -NCPU=$task.cpus \\
+        -MAXMEM=$max_memory \\
+        -cdnaseq $fasta
+
+    find \\
+        dropset \\
+        -type f \\
+        -exec sh -c 'mv "\$1" "\$(echo \$1 | sed s/$simple_name/$prefix/)"' sh {} \\;
```

Comment:
What happens if simple_name == prefix?
I think that conda fails if you try and mv the file to itself.

Reply:
Good point @SPPearce 

I have added a conditional to cater to such cases.